### ISSUE TITLE
[Bug]: --enable-return-routed-experts is rejected on TPU — AuxOutput connector requires Model Runner V2, which requires Triton

### ISSUE THREAD

**Issue by lk-chen** [vllm-project/tpu-inference#3636]
## Summary

`--enable-return-routed-experts` (R3) cannot be enabled on TPU as of vLLM `0b7f11a1eece3dc9c689f1b5ab90d60c03b10793` ([vllm-project/vllm#45635](https://github.com/vllm-project/vllm/pull/45635), *[AuxOutput] Add block-keyed storage for routed-expert outputs*). The engine is rejected at config validation, before any tpu-inference code runs.

The block is a single upstream gate. With that one gate relaxed, R3 works end to end on TPU — the full `tests/runner/test_moe_expert_ids.py` suite passes on v6e (evidence below).

## The blocker

`VllmConfig._verify_aux_output_compatibility()`  requires Model Runner V2:

https://github.com/vllm-project/vllm/blob/bc162b3f9280c2dc05dfd5aa282d29551f8f6d33/vllm/config/vllm.py#L1095-L1099

TPU can never satisfy this. Setting `VLLM_USE_V2_MODEL_RUNNER=1` trades one error for another, because MRV2 requires Triton :

https://github.com/vllm-project/vllm/blob/bc162b3f9280c2dc05dfd5aa282d29551f8f6d33/vllm/config/vllm.py#L3184-L3185

and Triton has no active driver on TPU, so the auto path falls back:

```
WARNING [vllm.py:732] Model Runner V2 requires Triton; using the V1 model runner instead.
```

There is no configuration that gets a TPU engine past this check.

## Why the gate exists

vLLM#45635 did not merely gate R3 behind MRV2 — it removed R3 from the V1 model runner (−127 lines in `vllm/v1/worker/gpu_model_runner.py`) and re-homed the data plane in `vllm/v1/worker/gpu/model_runner.py` plus the new `vllm/distributed/aux_output_connector/worker.py`.

That worker connector is not reusable off MRV2: `prepare_output()` takes `vllm.v1.worker.gpu.input_batch.InputBatch`, and it binds a capturer into `torch.nn.Module`s. tpu-inference runs a V1-style runner, so it needs its own worker-side implementation regardless.

The scheduler half (`AuxOutputSchedulerConnector`, `vllm/distributed/aux_output_connector/connector.py`) is runner-agnostic, which is what makes a TPU-side implementation viable.

## Related API changes in the same PR

These break the downstream R3 code but are straightforward ports, not blockers:

- `ModelRunnerOutput.routed_experts` (`RoutedExpertsLists`) → `aux_output_connector_output: dict[str, AuxRequestOutput]` (`vllm/v1/outputs.py:299`), consumed via `AuxOutputSchedulerConnector.take_output` (`vllm/v1/core/sched/scheduler.py:2075`).
- `ModelConfig.enable_return_routed_experts` → `AuxOutputConfig.enable_return_routed_experts` (`vllm/config/aux_output.py:14`).
- `get_routed_experts_attn_gid` removed from `vllm/model_executor/layers/fused_moe/routed_experts_capturer.py`.
- New requirement: R3 now raises `AuxOutput Connector requires prefix caching.` — it can no longer run with `enable_prefix_caching=False`. Also new: generate-runner-only and MoE-only checks. (The PP > 1 / DCP-PCP > 1 / KV-connector rejections are not new; they moved into this function from `_verify_args`.)

## Evidence that the gate is the only blocker

Patching only the gate to let non-GPU platforms through — the sole modification to the vLLM checkout, +5/−1 in `vllm/config/vllm.py`:

```python
if not self.use_v2_model_runner and not current_platform.is_tpu():
    raise ValueError(...)
```

then running the R3 suite on v6e:

```
$ python3 -m pytest -s -v tests/runner/test_moe_expert_ids.py

tests/runner/test_moe_expert_ids.py::TestMoEExpertIds::test_moe_expert_ids_returned_when_enabled PASSED
tests/runner/test_moe_expert_ids.py::TestMoEExpertIds::test_moe_expert_ids_not_returned_when_disabled PASSED
tests/runner/test_moe_expert_ids.py::TestMoEExpertIds::test_moe_expert_ids_batch_isolation PASSED
tests/runner/test_moe_expert_ids.py::TestMoEExpertIds::test_moe_expert_ids_sync_path_prompt_populated PASSED

================== 4 passed, 19 warnings in 533.08s (0:08:53) ==================
```

The engine logs confirm it ran on the V1 runner (`Model Runner V2 requires Triton; using the V1 model runner instead`) and still produced correct routed experts through `aux_output_connector_output`, on both the async path and the sync path (`async_scheduling=False` with `enable_continue_decode`).

**Caveat on scope:** single-chip v6e, `load_format="dummy"`, `tensor_parallel_size=1`, `Qwen/Qwen1.5-MoE-A2.7B`. Multi-chip TP/EP and DP are not exercised by this run, so this shows the gate is the only blocker for that configuration, not for every configuration.

## Proposed fix

A platform hook in `_verify_aux_output_compatibility` — allow a platform that supplies its own aux-output worker to bypass the `use_v2_model_runner` check — so the TPU worker implementation can be reached. A change in the gate alone is not enough for platforms without their own worker; it just exposes the missing data plane.

## Current CI status

`tests/runner/test_moe_expert_ids.py` runs only in `.buildkite/rl/moe_expert_ids.yml` (the main JAX pipeline `--ignore`s it, `.buildkite/pipeline_jax.yml:131`). That step is `soft_fail: true`, so on an unpatched pin it goes red and records a failure for "MoE Expert IDs" in the rl support matrix without blocking the build.

## Environment

- tpu-inference: `de1e6ff1e`
- vLLM: `0b7f11a1eece3dc9c689f1b5ab90d60c03b10793` (2026-09-21)
- TPU: v6e-8, `tpu_type=v6e-8`, 8 chips (1 used)
